api/drivers/aws/driver.js

```diff
@@ -255,6 +299,71 @@ class AWSDriver extends Driver {
       });
     });
   }
+
+  /**
+   * Calcul credit used by a user
+   *
+   * @method getUserCredit
+   * @return {Promise}
+   */
+  getUserCredit(user) {
+    let product = [];
+    let pri = [];
+    let duration = [];
+    var finalPrice = 0;
+
+    return new Promise((resolve, reject) => {
+      ConfigService.get('price')
+        .then((conf) => {
+          var jsonPrice = conf.price;
+          UserService.getUserHistory(user, 'aws')
```

Comment:
If you need to fetch to informations do not use two chained promises. Prefer `Promise.props` instead : http://bluebirdjs.com/docs/api/promise.props.html